google/cloud/forseti/notifier/notifier.py

```diff
@@ -136,70 +138,69 @@ def run(inventory_index_id, progress_queue, service_config=None):
     global_configs = service_config.get_global_config()
     notifier_configs = service_config.get_notifier_config()
 
-    if not inventory_index_id:
-        with service_config.scoped_session() as session:
+    violations = None
+    with service_config.scoped_session() as session:
+        if not inventory_index_id:
             inventory_index_id = (
                 DataAccess.get_latest_inventory_index_id(session))
-
-    # get violations
-    violation_access_cls = scanner_dao.define_violation(
-        service_config.engine)
-    violation_access = violation_access_cls(service_config.engine)
-    service_config.violation_access = violation_access
-    violations = violation_access.list(inventory_index_id)
-
-    violations = convert_to_timestamp(violations)
-
-    violations_as_dict = []
-    for violation in violations:
-        violations_as_dict.append(
-            scanner_dao.convert_sqlalchemy_object_to_dict(violation))
-
-    violations = scanner_dao.map_by_resource(violations_as_dict)
-
-    for retrieved_v in violations:
-        log_message = ('Retrieved {} violations for resource \'{}\''.format(
-            len(violations[retrieved_v]), retrieved_v))
-        LOGGER.info(log_message)
-        progress_queue.put(log_message)
-
-    # build notification notifiers
-    notifiers = []
-    for resource in notifier_configs['resources']:
-        if violations.get(resource['resource']) is None:
-            log_message = 'Resource \'{}\' has no violations'.format(
-                resource['resource'])
-            progress_queue.put(log_message)
+        scanner_index_id = scanner_dao.get_latest_scanner_id(
+            session, inventory_index_id)
+        if not scanner_index_id:
+            LOGGER.warn('No scanner index found.')
```

Comment:
The notifier should halt here if there are no valid `scanner_index_id` found.  Otherwise, this would end up sending other violations back that's not relevant to the current forseti run.